action/protocol/staking/handlers.go

```diff
@@ -127,16 +127,24 @@ func (p *Protocol) handleUnstake(ctx context.Context, act *action.Unstake, csm C
 		return p.settleAction(ctx, csm, uint64(fetchErr.failureStatus), gasFee)
 	}
 
-	// update bucket
-	bucket.UnstakeStartTime = blkCtx.BlockTimeStamp
-	if err := updateBucket(csm, act.BucketIndex(), bucket); err != nil {
-		return nil, errors.Wrapf(err, "failed to update bucket for voter %s", bucket.Owner)
+	if blkCtx.BlockTimeStamp.Before(bucket.StakeStartTime.Add(bucket.StakedDuration)) {
+		err := fmt.Errorf("bucket is not ready to be unstaked, current time %s, required time %s",
+			blkCtx.BlockTimeStamp, bucket.StakeStartTime.Add(bucket.StakedDuration))
+		log.L().Debug("Error when withdrawing bucket", zap.Error(err))
+		return p.settleAction(ctx, csm, uint64(iotextypes.ReceiptStatus_ErrUnstakeBeforeMaturity), gasFee)
 	}
 
 	candidate := csm.GetByOwner(bucket.Candidate)
```

Comment:
can you move this before check unstake time?

in other handlers, we always check candidate exist/conflict first, before we check bucket, let's be consistent